src/cli/utils/utils.go

```diff
@@ -32,14 +32,25 @@ func GetAccountAndConnect(ctx context.Context) (repository.Repository, *account.
 
 	repoID := cfg.RepoID
 	if len(repoID) == 0 {
-		repoID = "not_found"
+		repoID = events.RepoID
 	}
 
 	r, err := repository.Connect(ctx, cfg.Account, cfg.Storage, repoID, options.Control())
 	if err != nil {
 		return nil, nil, clues.Wrap(err, "Failed to connect to the "+cfg.Storage.Provider.String()+" repository")
 	}
 
+	s3Config, errS3Config := cfg.Storage.S3Config()
+	m365Config, errm365Config := cfg.Account.M365Config()
+
+	// repo config is already set while repo connect and init. This is just to confirm correct values.
+	// So won't fail is the write fails
+	err = config.WriteRepoConfig(ctx, s3Config, m365Config, r.GetID())
```

Comment:
Writes occuring within a Get func creates a side effect and a smell.  It will be much cleaner if the two behaviors are separated, giving the caller clear protection against accidental mutation.  That same caller can handle overwriting the value as needed.  Or alternatively, a higher level func can replace the get call, and the get and write can be handled separately.